test/evergreen/code_coverage/code_coverage_utils.py

```diff
@@ -29,87 +29,100 @@
-
-        build_dirs.append(this_build_dir)
+        task_bucket_info.append({'build_dir': this_build_dir, 'task_bucket': []})
 
+        # Check build dir for coverage files.
         found_compile_time_coverage_files = False
         found_run_time_coverage_files = False
-
-        # Check build dir for coverage files
-        for root, dirs, files in os.walk(this_build_dir):
+        for _, _, files in os.walk(this_build_dir):
             for filename in files:
                 if filename.endswith('.gcno'):
                     found_compile_time_coverage_files = True
                 if filename.endswith('.gcda'):
                     found_run_time_coverage_files = True
 
         logging.debug('Found compile time coverage files in {} = {}'.
-              format(this_build_dir, found_compile_time_coverage_files))
+            format(this_build_dir, found_compile_time_coverage_files))
         logging.debug('Found run time coverage files in {}     = {}'.
-              format(this_build_dir, found_run_time_coverage_files))
+            format(this_build_dir, found_run_time_coverage_files))
 
-        if not setup and not found_compile_time_coverage_files:
+        if not found_compile_time_coverage_files:
             sys.exit('No compile time coverage files found within {}. Please build for code coverage.'
-                     .format(this_build_dir))
+                    .format(this_build_dir))
 
-    logging.debug("Build dirs: {}".format(build_dirs))
+    logging.debug("Build dirs: {}".format(task_bucket_info))
 
-    return build_dirs
+    return task_bucket_info
+
+# Create the relevant build directories to run tasks.
+def setup_build_dirs(build_dir_base, parallel, setup_task_list):
+    task_bucket_info = list()
+
+    base_build_dir = "{}{}".format(build_dir_base, 0)
+    if os.path.exists(base_build_dir):
+        sys.exit('build directory exists within {}.'.format(base_build_dir))
+
+    logging.debug('Creating build directory {}.'.format(base_build_dir))
+    os.mkdir(base_build_dir)
+
+    for build_num in range(parallel):
+        this_build_dir = "{}{}".format(build_dir_base, build_num)
+        task_bucket_info.append({'build_dir': this_build_dir, 'task_bucket': []})
+
+    logging.debug("Build dirs: {}".format(task_bucket_info))
+
+    logging.debug("Compiling base build directory: {}".format(base_build_dir))
+    start_time = datetime.now()
+    for task in setup_task_list:
+        try:
+            p = PushWorkingDirectory(base_build_dir)
+            split_command = task.split()
+            subprocess.run(split_command, check=True, capture_output=True)
+            p.pop()
+        except subprocess.CalledProcessError as exception:
+            logging.error(f'Command {exception.cmd} failed with error {exception.returncode}')
+    end_time = datetime.now()
+    diff = end_time - start_time
+
+    logging.debug("Finished setup and took {} seconds".format(diff.total_seconds()))
```

Comment:
I think it is valuable to measure the build time here, especially since it runs for 2 minutes now. If we can reduce that down in the future it would be great.